precompile/contracts/warp/config.go

```diff
@@ -202,7 +208,7 @@ func (c *Config) VerifyPredicate(predicateContext *precompileconfig.PredicateCon
 		context.Background(),
 		&warpMsg.UnsignedMessage,
 		predicateContext.SnowCtx.NetworkID,
-		warpValidators.NewState(predicateContext.SnowCtx), // Wrap validators.State on the chain snow context to special case the Primary Network
+		warpValidators.NewState(predicateContext.SnowCtx, func() bool { return c.RequirePrimaryNetworkSigners }), // Wrap validators.State on the chain snow context to special case the Primary Network
```

Comment:
I think we need to make a distinction between the `sourceBlockchainID` of the unsigned message here.

The reason is that validator management related messages from the P-Chain will always need to be signed by only the receiving L1's validator set, because P-Chain validators that do not track the given L1 will not be willing to sign the message for it. Given that all validators still need to track the P-Chain, this should be possible for now.

The precompile configuration flag should only apply for messages where the `sourceChainID` is the C-Chain I believe.

Reply:
made a special case for the P-Chain per this comment.
Additionally, there is an alternative implementation we can consider: https://github.com/ava-labs/subnet-evm/pull/1232/files